jetty-http/src/test/java/org/eclipse/jetty/http/QuotedQualityCSVTest.java

```diff
@@ -309,5 +312,44 @@ public void testNoQuality()
         values.addValue("one,two;,three;x=y");
         Assert.assertThat(values.getValues(),Matchers.contains("one","two","three;x=y"));
     }
+
+
+    @Test
+    public void testQuality()
+    {
+        List<String> results = new ArrayList<>();
+
+        QuotedQualityCSV values = new QuotedQualityCSV()
+        {
+            @Override
+            protected void parsedValue(StringBuffer buffer)
+            {
+                super.parsedValue(buffer);
+                results.add("parsedValue: " + buffer.toString());
+            }
+
+            @Override
+            protected void parsedParam(StringBuffer buffer, int valueLength, int paramName, int paramValue)
+            {
+                String param = buffer.substring(paramName, buffer.length());
+                results.add("parsedParam: " + param);
+                super.parsedParam(buffer, valueLength, paramName, paramValue);
+            }
+        };
+
+
+        /*
+        The quality value q=0.5 parameter is treated specially, it is consumed and used as the weight for an empty string value
+        however the parameter p=0.4 is not consumed and is instead added to values
+
+        Note: allowing the q parameter by itself is not specified behaviour in the RFC, but the implementation here is lenient
+        */
```

Comment:
This comment kind of says it all.... but I'd say it around the other way....

- the provided string is not legal according to some RFCs (not a token because of = and not a parameter because not preceeded by ;
- the provided string is legal according to RFC ???? which allows for just parameters (or pairs or whatever it calls them)
- So the QuotedCSV implementation is lenient and adopts the later interpretation and thus sees q=0.5 and p=0.5 both as parameters 
- However the QuotedQualityCSV only handles the q parameter and that is consumed from the parameter string.  The p parameter is neither handled nor consumed and is effectively ignored